jena-base/src/main/java/org/apache/jena/atlas/lib/cache/CacheSimple.java

```diff
@@ -37,120 +37,81 @@
 
 public class CacheSimple<K,V> implements Cache<K,V>
 {
-    private final V[] values ; 
-    private final K[] keys ;
-    private final int size ;
-    private int currentSize = 0 ;
-    private BiConsumer<K,V> dropHandler = null ;
+	private Map<K,V> internalCache;
+	
```

Comment:
Just to be clear, am I correct to be thinking that a `CacheSimple` should never expand beyond the size of its constructor argument? I'm asking because that's exactly what my new test checked, and it passes for my `LinkedHashMap` impl, but not for the original. I build a `CacheSimple<Integer>(5)` and put 10 things in it, mapped from 10 different keys each to the same value, and then its `size()` is 10 when it should only be 5. Or am I misunderstanding the policy here?